array.go

```diff
@@ -1022,3 +1023,123 @@ func countElement(array []interface{}, element interface{}) int {
 	}
 	return count
 }
+
+// IsOrdered succeeds if array is ordered based on optional `comparator` function.
+// For default, it will use built-in comparator function for each data type.
+// It requires all elements in the array to have same data type
+//
+// Example:
+//
+//	array := NewArray(t, []interface{}{100, 101, 102})
+//	array.IsOrdered(func(x, y *Value) bool {
+//		valX, _ := x.Raw().(int)
+//		valY, _ := y.Raw().(int)
+//		return valX >= valY
+//	})
+func (a *Array) IsOrdered(less ...func(x, y *Value) bool) *Array {
+	a.chain.enter("IsOrdered()")
+	defer a.chain.leave()
+
+	if a.chain.failed() {
+		return a
+	}
+
+	if len(less) > 1 {
+		a.chain.fail(AssertionFailure{
+			Type: AssertUsage,
+			Errors: []error{
+				errors.New("unexpected multiple less arguments"),
+			},
+		})
+		return a
+	}
+
+	var fn func(x, y *Value) bool
+	if len(less) == 1 {
+		fn = less[0]
+	} else {
+		// use default comparator based on data type
```

Comment:
I suggest to extract default comparison algorithm into a separate function, like:

```
func defaultCompare(x, y *Value) bool
```